packages/main/cypress/specs/RadioButton.cy.tsx

```diff
@@ -0,0 +1,497 @@
+
+import RadioButton from "../../src/RadioButton.js";
+import Input from "../../src/Input.js";
+import Label from "../../src/Label.js";
+import Switch from "../../src/Switch.js";
+import Title from "../../src/Title.js";
+
+describe("Rendering", () => {
+	it("DOM structure", () => {
+		cy.mount(
+			<>
+				<RadioButton id="testRbtn" text="N/A" disabled name="test"></RadioButton>
+				<RadioButton id="testRbtn1" text="Male" name="test"></RadioButton>
+				<RadioButton id="groupRbReadOnly" name="a" readonly text="Option E"></RadioButton>
+			</>);
+
+		cy.get("#testRbtn1")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("not.have.attr", "aria-disabled");
+
+		cy.get("#testRbtn1")
+			.shadow()
+			.find("input")
+			.should("not.have.attr", "disabled");
+
+		cy.get("#testRbtn")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("have.attr", "aria-disabled");
+
+		cy.get("#testRbtn")
+			.shadow()
+			.find("input")
+			.should("have.attr", "disabled");
+
+		cy.get("#groupRbReadOnly")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("have.attr", "aria-disabled");
+
+		cy.get("#groupRbReadOnly")
+			.shadow()
+			.find("input")
+			.should("have.attr", "readonly");
+
+		cy.get("#groupRbReadOnly")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("have.attr", "aria-disabled");
+	});
+});
+
+describe("RadioButton general interaction", () => {
+	it("tests change event", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(<RadioButton id="rb1" onChange={changeEventStub}></RadioButton>);
+
+		cy.get("#rb1")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb1").realClick();
+		cy.wrap(changeEventStub)
+			.should("have.been.called");
+
+		cy.get("#rb1").realClick();
+		cy.wrap(changeEventStub)
+			.should("have.been.calledOnce");
+	});
+
+	it("tests change event upon ENTER", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(
+			<>
+				<RadioButton id="rb1"></RadioButton>
+				<RadioButton id="rb2" text="Option B" onChange={changeEventStub}></RadioButton>
+			</>);
+
+		cy.get("#rb1")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb1").realClick();
+		cy.get("#rb1").realPress("Tab");
+
+		cy.get("#rb2")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb2").realPress("Enter");
```

Comment:
I would add assertion that change was not triggered before enter key. Do this to all tests related to the change event

```
		cy.wrap(changeEventStub)
			.should("not.have.been.called");
```